File: pytools/examples/lua53/server/protocols/AckSceneEnter.lua
AckSceneEnter = {}
setmetatable(AckSceneEnter, {__index = _G})
local _ENV = AckSceneEnter


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckSceneEnter})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.player = nil
end


function encode(self)
	pack = packet.new()
	pack:writeBytes(self.player:getBytes())
	return pack:encode(Msg.P_ACK_SCENE_ENTER)
end


function setPlayer(self, player)
	self.player = player
end

File: pytools/data/data.code.php.server/AckSceneEnter.php
<?php
namespace protocol;

use network\Packet;


class AckSceneEnter
{
	private $player = null;


	public function encode()
	{
		$packet = new Packet();
		$packet->writeBytes($this->player->getBytes());
		return $packet->encode(Msg::$P_ACK_SCENE_ENTER);
	}


	public function setPlayer($player)
	{
		$this->player = $player;
	}

}
